### PR TITLE
Improving connectivity page added

### PR DESCRIPTION
**Pull request by har00ga**
Solves #178 

This PR includes an initial draft for an 'improving connectivity' page in the `/mine/` section of the docs. Was not sure how much detail I should go into as far as instructions, but happy to expand on things if preferred

### CODE REVIEW

File: docs/mine/connectivity.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+title: Improving connectivity
+description: Tips and tricks for improving a miner's connectivity to the Filecoin network.
+---
+
+
+# Improving connectivity
+
+Filecoin miners, like participants in all P2P protocols, require a steady and quality pool of peers to communicate with in order to perform their various functions. For other participants on the network to establish incoming P2P connections with a miner, a few conditions must be met:
```
Comment (mishmosh) [inline docs/mine/connectivity.md:9]:
```suggestion
Filecoin miners, like participants in all peer-to-peer protocols, require a steady and quality pool of peers to communicate with in order to perform their various functions. For other participants on the network to establish incoming P2P connections with a miner, a few conditions must be met:
```
Spell out peer-to-peer the first time it's used on a page.

File: docs/mine/connectivity.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+It is important to ensure that the public IP address as well as the port that was configured above is listed in the miner configuration file's `AnnounceAddresses` address list. DNS4 multi-address or IPV6 formats are also valid.
+
+Below is an example configuration file in which the public IP address is `1.2.3.4`:
+
+```
+[Libp2p]
+   ListenAddresses = ["/ip4/0.0.0.0/tcp/5472"]
+   AnnounceAddresses = ["/ip4/1.2.3.4/tcp/10240"]
+In the example port 10240 is forwarded to <internal-miner-host-ip>:5472
```
Comment (mishmosh) [inline docs/mine/connectivity.md:51]: Missing some quotes & unquotes. I made a suggestion below but please double-check that Github is parsing it correctly. The preformatted text should end after line 50, and the 2 key values in line 51 should be in backticks.

```suggestion
```
In this example, port `10240` is forwarded to `<internal-miner-host-ip>:5472`.
```

File: docs/mine/connectivity.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Miners can verify that things are working by running by using telnet eg: telnet 1.2.3.4 10240 (nc also works)
+Miners should see a plaintext  `/multistream/1.0.0` in the response.
+```
```
Comment (mishmosh) [inline docs/mine/connectivity.md:54]:
```suggestion
```

File: docs/mine/connectivity.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+It is important to ensure that the public IP address as well as the port that was configured above is listed in the miner configuration file's `AnnounceAddresses` address list. DNS4 multi-address or IPV6 formats are also valid.
+
+Below is an example configuration file in which the public IP address is `1.2.3.4`:
+
+```
+[Libp2p]
+   ListenAddresses = ["/ip4/0.0.0.0/tcp/5472"]
+   AnnounceAddresses = ["/ip4/1.2.3.4/tcp/10240"]
+In the example port 10240 is forwarded to <internal-miner-host-ip>:5472
+Miners can verify that things are working by running by using telnet eg: telnet 1.2.3.4 10240 (nc also works)
```
Comment (mishmosh) [inline docs/mine/connectivity.md:52]:
```suggestion
Miners can verify that things are working by running by using telnet: `telnet 1.2.3.4 10240` (nc also works).
```

File: docs/mine/connectivity.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+3. Find the section to configure port forwarding
+4. Choose a port, and configure a port forwarding rule with the following values:
+    * External port: [CHOSEN PORT]
+    * Internal port: [CHOSEN PORT]
+    * Protocol: TCP
+    * IP Address: Private IP address of the host system running the miner
+
+## Setting a public IP address
+
+It is important to ensure that the public IP address as well as the port that was configured above is listed in the miner configuration file's `AnnounceAddresses` address list. DNS4 multi-address or IPV6 formats are also valid.
```
Comment (mishmosh) [inline docs/mine/connectivity.md:43]: Include a bit of context about why it's important. travisperson describes it this way: "To ensure quick and easy lookup for deal and retrievals..."

File: docs/mine/connectivity.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+```
+lotus-miner net peers
+``` 
+
+If a very low peer count is present (1-5), it is possible to manually connect the miner to the DHT by utilising one of the bootstrap peers listed in the branch's `./build/bootstrap/bootstrappers.pi` file.
+
+## Port forwarding
+
+It is highly recommended to use port forwarding for a miner's `libp2p` address. By default, this port is randomised; for optimal connectivity, ensure that it is set to a static IP.
```
Comment (mishmosh) [inline docs/mine/connectivity.md:28]: Include a bit of context about why it's highly recommended.

File: docs/mine/connectivity.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+# Improving connectivity
+
+Filecoin miners, like participants in all P2P protocols, require a steady and quality pool of peers to communicate with in order to perform their various functions. For other participants on the network to establish incoming P2P connections with a miner, a few conditions must be met:
+
+* The miner's public IP address must be known
+* The protocol (TCP/UDP) and port number (0-65535) on which the miner is listening must be known 
+* All routers & firewalls must be configured to allow incoming traffic on that protocol/port combination
+
+The below sections assist in performing these tasks.
```
Comment (mishmosh) [inline docs/mine/connectivity.md:15]: Stronger recommendation & context here:
```suggestion
The following steps are highly recommended for all miners who wish to successfully accept storage and retrieval deals.
```

File: docs/mine/connectivity.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+## Checking peer count
+To ensure storage and retrieval deals operate smoothly, it is recommended to regularly check how many peers a miner is connected to. In the Lotus client, a manual peer check can be performed with the command:
```
Comment (mishmosh) [inline docs/mine/connectivity.md:18]: how regularly? and is it regularly once the miner is connected, or just once?

File: docs/mine/connectivity.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+## Checking peer count
+
+To ensure storage and retrieval deals operate smoothly, it is recommended to check how many peers a miner is connected to after each start-up. In the Lotus client, a manual peer check can be performed with the command:
+
+```
+lotus-miner net peers
+``` 
+
+If a very low peer count is present (1-5), it is possible to manually connect the miner to the DHT by utilising one of the bootstrap peers listed in the branch's `./build/bootstrap/bootstrappers.pi` file.
```
Comment (arajasek) [inline docs/mine/connectivity.md:25]: It'd be good to include the command here (`lotus net connect <address from bootstrappers>`).

Comment (mishmosh) [reply docs/mine/connectivity.md:25]: Thanks for the suggestion @arajasek! Since this PR already merged, @har00ga can you address in a follow-up?

Comment (har00ga) [reply docs/mine/connectivity.md:25]: Can do!

**Pull request merged**